File: cakk-domain/mysql/src/main/java/com/cakk/domain/mysql/converter/VerificationStatusConverter.java
package com.cakk.domain.mysql.converter;

import java.util.stream.Stream;

import jakarta.persistence.AttributeConverter;

import com.cakk.common.enums.ReturnCode;
import com.cakk.common.enums.VerificationStatus;
import com.cakk.common.exception.CakkException;

public class VerificationStatusConverter implements AttributeConverter<VerificationStatus, Integer> {
	@Override
	public Integer convertToDatabaseColumn(VerificationStatus verificationStatus) {
		if (verificationStatus == null) {
			return null;
		}

		return verificationStatus.code;
	}

	@Override
	public VerificationStatus convertToEntityAttribute(Integer code) {
		if (code == null) {
			return null;
		}

		return Stream.of(VerificationStatus.values())
			.filter(verificationStatus -> verificationStatus.code == code)
			.findFirst()
			.orElseThrow(() -> new CakkException(ReturnCode.INTERNAL_SERVER_ERROR));
	}
}

File: cakk-infrastructure/persistence-mysql/src/main/java/com/cakk/infrastructure/persistence/converter/VerificationStatusConverter.kt
package com.cakk.infrastructure.persistence.converter

import com.cakk.common.enums.ReturnCode
import com.cakk.common.enums.VerificationStatus
import com.cakk.common.exception.CakkException
import jakarta.persistence.AttributeConverter
import java.util.stream.Stream

class VerificationStatusConverter : AttributeConverter<VerificationStatus?, Int?> {

	override fun convertToDatabaseColumn(verificationStatus: VerificationStatus?): Int? {
        return verificationStatus?.code
    }

    override fun convertToEntityAttribute(code: Int?): VerificationStatus? {
        return if (code == null) {
            null
        } else Stream.of(*VerificationStatus.values())
                .filter { verificationStatus: VerificationStatus -> verificationStatus.code == code }
                .findFirst()
                .orElseThrow { CakkException(ReturnCode.INTERNAL_SERVER_ERROR) }
    }
}

